plugins/EpadDB.js

```diff
@@ -2209,36 +2339,129 @@ async function epaddb(fastify, options, done) {
-            dock
+            // eslint-disable-next-line no-await-in-loop
+            await dock
               .checkContainerExistance(containerName)
+              // eslint-disable-next-line no-loop-func
               .then((resInspect) => {
                 if (resInspect.message === '404') {
                   fastify.log.info(
                     'not a real error. Container has not found so we can create new one'
                   );
-                  throw new Error('404');
+                  if (sequence) {
+                    nonseqresult.push(pluginObj);
+                    seqresult.push(pluginObj);
+                  } else {
+                    throw new Error('404');
+                  }
                 }
                 if (resInspect.State.Status !== 'running') {
                   fastify.log.info(`container is not running : ${containerName}`);
                   dock.deleteContainer(containerName).then((deleteReturn) => {
                     fastify.log.info(`delete container result :${deleteReturn}`);
-                    result.push(pluginObj);
-                    fastify.runPluginsQueueInternal(result, request);
+                    nonseqresult.push(pluginObj);
+                    seqresult.push(pluginObj);
+                    if (!sequence) {
+                      fastify.runPluginsQueueInternal(nonseqresult, request);
+                    }
                   });
                 }
               })
+              // eslint-disable-next-line no-loop-func
               .catch((err) => {
                 fastify.log.info(`inspect element err : ${err}`);
                 if (err.message === '404') {
-                  result.push(pluginObj);
-                  fastify.runPluginsQueueInternal(result, request);
+                  nonseqresult.push(pluginObj);
+                  seqresult.push(pluginObj);
+                  if (!sequence) {
+                    fastify.runPluginsQueueInternal(nonseqresult, request);
+                  }
                 }
               });
+          }
+          if (sequence) {
+            const removeIds = [];
+            for (let i = 0; i < seqresult.length; i += 1) {
+              if (!globalMapQueueById.has(seqresult[i].id)) {
+                globalMapQueueById.set(seqresult[i].id, '');
+              } else {
+                removeIds.push(seqresult[i].id);
+              }
+            }
+
+            for (let i = 0; removeIds.length; i += 1) {
+              for (let k = 0; seqresult.length; k += 1) {
+                if (seqresult[k].id === removeIds[i]) {
+                  seqresult[k] = { id: -1 };
+                }
+              }
+            }
+            await fastify.runPluginsQueueInternal(seqresult, request);
+          }
+        });
+    } catch (err) {
+      fastify.log.error(`runPluginsQueue error : ${err}`);
+    }
+  });
+
+  fastify.decorate('getNextPluginInSubQueue', async (paramQid, request) => {
```

Comment:
this doesn't return a plugin id or plugin info. I'm guessing it triggers plugin. please put a comment on what it does